### PR TITLE
Tests: Remove queue overrides / limit concurrency

### PR DESCRIPTION
**Pull request by jaymell**
Ensure all backend variations are properly end-to-end tested
by removing in-memory queue override; this currently
requires limiting test concurrency to 1 because tests are
currently sharing backend services.

### CODE REVIEW

File: server/run-tests.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Run tests with various configurations:
 
-TEST_COMMAND="cargo test --all --all-features --all-targets"
+TEST_COMMAND="cargo test --all --all-features --all-targets -- --test-threads=1"
```
Comment (tasn) [inline server/run-tests.sh:5]: I could have sworn we had this in already. 🤔

File: server/testing-queue.conf [hunk 2]
```diff
@@ -0,0 +1,17 @@
+include classpath("application.conf")
```
Comment (tasn) [inline server/testing-queue.conf:1]: What dis?

File: server/run-tests.sh (hunk 1 above)
Comment (jaymell) [reply server/run-tests.sh:5]: It was added, then removed in later commit.

File: server/testing-queue.conf (hunk 2 above)
Comment (jaymell) [reply server/testing-queue.conf]: Something that shouldn't have been added. Removed.

File: server/run-tests.sh (hunk 1 above)
Comment (tasn) [reply server/run-tests.sh:5]: Ah cool.

**Review by tasn — APPROVED** [review]

**Pull request merged**